### PR TITLE
front: highlight unindentified prs in itinerary component

### LINKED ISSUE #8652: Identify unrecognized PR in the infrastructure
### Description and goal

In Operational Studies > Route, use the endpoint created in issue #8636 to:
- Mark unrecognized PRs in red in the list.
- Ensure that valid PRs are displayed on the map.

Part of: https://github.com/osrd-project/osrd-confidential/issues/632

### Acceptance criteria

- [ ]  Unrecognized PRs are now displayed in red in the Itinerary component list.
- [ ]  Unrecognized PRs are no longer visible on the map.
- [ ] The number of each PR on the map remains the same, even if some PRs are withdrawn.
- [ ]  Valid PRs are still displayed in black.
- [ ]  If the origin is not recognized, it is not replaced by the next point on the map.

### PR DESCRIPTION
**Pull request by Caracol3**
close #8652

### CODE REVIEW

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Destination.tsx [hunk 1]
```diff
@@ -8,16 +9,23 @@ import { useOsrdConfActions, useOsrdConfSelectors } from 'common/osrdContext';
...
   const destination = useSelector(getDestination);
 
   const dispatch = useDispatch();
   const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
 
+  const isInvalid =
+    invalidPathItems &&
+    destination &&
+    'trigram' in destination &&
+    invalidPathItems.includes(destination.trigram);
```
Comment (emersion) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Destination.tsx]: Could we avoid using the `trigram` to identify path items? Path items all have an ID (note, _not_ the OP ID, which is stored separately in `operational_point`).

**Comment by emersion** [comment]
This will need rebasing on top of https://github.com/OpenRailAssociation/osrd/pull/9289 (sorry!).

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx [hunk 2]
```diff
@@ -12,9 +13,10 @@ import { makeEnumBooleans } from 'utils/constants';
 
 type OriginProps = {
   zoomToFeaturePoint: (lngLat?: Position, id?: string) => void;
+  invalidPathItems?: string[];
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx]: I think it would be more elegant to give isInvalid to each component (Origin, Via and Destination) instead of the whole array invalidPathItems. 
You can create a utils function to check whether a pathItem is invalid or not and then call it in Itinerary.
```suggestion
  isInvalid?: boolean;
```

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx [hunk 3]
```diff
@@ -73,7 +78,7 @@ const extractMarkerInformation = (pathSteps: (PathStep | null)[], showStdcmAsset
           type: MARKER_TYPE.VIA,
           marker: cur,
           imageSource: showStdcmAssets ? stdcmVia : viaSVG,
-          index,
+          index: index,
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx]:
```suggestion
          index,
```

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 4]
```diff
@@ -177,9 +177,24 @@ export const usePathfinding = (
...
+
   const generatePathfindingParams = (): PostInfraByInfraIdPathfindingBlocksApiArg | null => {
     setPathProperties?.(undefined);
-    return getPathfindingQuery({ infraId, rollingStock, origin, destination, pathSteps });
+
+    const filteredPathSteps = pathSteps.filter(
+      (step) =>
+        step !== null &&
+        step.coordinates !== null &&
+        !('trigram' in step && invalidItems.includes(step.trigram))
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: I think this is enough :)
```suggestion
        pathStepIsInvalid(step, invalidItems)
```

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx [hunk 5]
```diff
@@ -57,9 +58,13 @@ const formatPointWithNoName = (
...
-const extractMarkerInformation = (pathSteps: (PathStep | null)[], showStdcmAssets: boolean) =>
-  pathSteps.reduce((acc, cur, index) => {
-    if (cur && cur.coordinates) {
+const extractMarkerInformation = (
+  pathSteps: (PathStep | null)[],
+  showStdcmAssets: boolean,
+  invalidItems: string[] = []
+) => {
+  return pathSteps.reduce((acc, cur, index) => {
+    if (cur && cur.coordinates && (!('trigram' in cur) || !invalidItems.includes(cur.trigram))) {
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx]: This condition is enough: if you don't find the pathItem in the infra, you won't have any coordinates. You can remove invalidItems from this component
```suggestion
    if (cur && cur.coordinates) {
```

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx (hunk 2 above)
Comment (emersion) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx]: I think we don't have `PathStep`s objects at this point. If one `PathItemLocation` is invalid, we don't have a pathfinding result. We only have `TrainScheduleBase.paths` and nothing else IIRC.

Comment (emersion) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx]: The above seems to be false: the frontend strips invalid `PathItemLocation` entries and re-laucnhes pathfinding. So in the end we do have `PathStep`s.

Related: https://github.com/OpenRailAssociation/osrd/pull/9270#discussion_r1806605475

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx [hunk 6]
```diff
@@ -37,7 +45,7 @@ const Vias = ({ zoomToFeaturePoint, shouldManageStopDuration }: ViasProps) => {
       <Droppable droppableId="droppableVias">
         {(provided) => (
           <div {...provided.droppableProps} ref={provided.innerRef}>
-            {vias.map((via, index) => (
+            {pathSteps.map((via, index) => (
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: Why do you use pathSteps instead of vias here ?

Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: Why do you use pathSteps instead of vias here ?

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx [hunk 7]
```diff
@@ -12,13 +12,21 @@ import { useAppDispatch } from 'store';
...
   zoomToFeaturePoint: (lngLat?: Position, id?: string) => void;
+  pathSteps: PathStep[];
+  isPathStepInvalid: (step: PathStep | null) => boolean;
 };
 
-const Vias = ({ zoomToFeaturePoint, shouldManageStopDuration }: ViasProps) => {
+const Vias = ({
+  zoomToFeaturePoint,
+  shouldManageStopDuration,
+  pathSteps,
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: No need to give pathSteps in this component, you can directly use the vias (from the selector getVias)

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx [hunk 8]
```diff
@@ -57,9 +57,9 @@ const formatPointWithNoName = (
   </>
 );
 
-const extractMarkerInformation = (pathSteps: (PathStep | null)[], showStdcmAssets: boolean) =>
-  pathSteps.reduce((acc, cur, index) => {
-    if (cur && cur.coordinates) {
+const extractMarkerInformation = (pathSteps: (PathStep | null)[], showStdcmAssets: boolean) => {
+  return pathSteps.reduce((acc, cur, index) => {
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx]: You can remove the return here and the brackets ;)

File: front/src/styles/scss/common/components/_pathfinding.scss [hunk 9]
```diff
@@ -91,6 +91,12 @@
     }
   }
 }
+.via,
+.place {
+  &.invalid-path-item {
+    color: #e05206;
```
Comment (clarani) [inline front/src/styles/scss/common/components/_pathfinding.scss]: Do you think you can use a color from the theme ? You can use theme('colors.orange') for instance (if Thibaut defined an orange color), or directly var(--orange)

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 10]
```diff
@@ -225,8 +236,24 @@ export const usePathfinding = (
...
+              .map((item) => {
+                return 'trigram' in item.path_item ? item.path_item.trigram : null;
+              })
+              .filter((trigram): trigram is string => trigram !== null);
 
-          if (
+            if (invalidPathItems.length > 0) {
+              const updatedPathSteps = pathSteps.map((step) => {
+                if (step && 'trigram' in step && invalidPathItems.includes(step.trigram)) {
+                  return { ...step, invalid: true };
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Is it invalid or isInvalid ?

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 11]
```diff
@@ -225,8 +236,24 @@ export const usePathfinding = (
 
         try {
           const pathfindingResult = await postPathfindingBlocks(pathfindingInput).unwrap();
+          if (pathfindingResult.status === 'invalid_path_items') {
+            const invalidPathItems = pathfindingResult.items
+              .map((item) => {
+                return 'trigram' in item.path_item ? item.path_item.trigram : null;
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Can you add a comment TODO here to precise that we currently handle only invalid pathSteps with trigram ? Later we will have to handle invalid pathSteps with trackOffset, opId and uic

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 12]
```diff
@@ -225,8 +236,24 @@ export const usePathfinding = (
...
+            if (invalidPathItems.length > 0) {
+              const updatedPathSteps = pathSteps.map((step) => {
+                if (step && 'trigram' in step && invalidPathItems.includes(step.trigram)) {
+                  return { ...step, invalid: true };
+                }
+                return step;
+              });
+              dispatch(updatePathSteps({ pathSteps: updatedPathSteps }));
+              pathfindingDispatch({ type: 'PATHFINDING_PARAM_CHANGED' });
+            }
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Maybe you can dispatch here a message for the user which explains that the pathfinding has failed because some pathSteps were not found on the infra ?
We should display this message even if invalidPathSteps is empty (as it contains only the invalidPathSteps with trigram and not all the invalidPathSteps)

**Review by clarani — COMMENTED** [review]
Super nice work :) almost there

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 10 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: My bad. It is indeed 'isInvalid'.

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx (hunk 8 above)
Comment (Caracol3) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx]: done

File: front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx (hunk 5 above)
Comment (Caracol3) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx]: done

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx (hunk 6 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: I struggled yesterday with the Vias component so had to do with the PathSteps but indeed, today (with a clearer mind 😁) there is no problem using vias.

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx (hunk 7 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: done

File: front/src/styles/scss/common/components/_pathfinding.scss (hunk 9 above)
Comment (Caracol3) [reply front/src/styles/scss/common/components/_pathfinding.scss]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 11 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 12 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: Now even if there is an invalid pathSteps, the pathfinding succeed (if there is at least two valid pathSteps).
Do we still want to dispatch a message then ?

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx (hunk 2 above)
Comment (clarani) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx]: It's good for me, @emersion ?

File: front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx [hunk 13]
```diff
@@ -49,6 +50,8 @@ const Itinerary = ({
   const { t } = useTranslation('operationalStudies/manageTrainSchedule');
   const { openModal } = useModal();
 
+  const isPathStepInvalid = (step: PathStep | null): boolean => step?.isInvalid || false;
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx]: Do you think you can move this function in a utils ?

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx [hunk 14]
```diff
@@ -16,9 +17,10 @@ import ViaStopDurationSelector from './ViaStopDurationSelector';
 type ViasProps = {
   shouldManageStopDuration?: boolean;
   zoomToFeaturePoint: (lngLat?: Position, id?: string) => void;
+  isInvalid: (step: PathStep | null) => boolean;
```
Comment (clarani) [inline front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: Since this function is in an utils, you can remove this props and directly use the function

Comment (Caracol3) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx]: done

File: front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx (hunk 13 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/components/Itinerary/Itinerary.tsx]: done

File: front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx (hunk 2 above)
Comment (emersion) [reply front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Origin.tsx]: Yeah, the new `isInvalid` field sgtm :)

File: front/package.json
Comment (clarani) [inline front/package.json:1]: Can you make this change in a separated commit ? You can call me if you don't know how to do this :)

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 15]
```diff
@@ -230,7 +241,38 @@ export const usePathfinding = (
             pathfindingResult.failed_status === 'pathfinding_not_found' &&
             pathfindingResult.error_type === 'incompatible_constraints';
 
-          if (pathfindingResult.status === 'success' || incompatibleConstraintsCheck) {
+          const isInvalidPathItems =
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Maybe it's a better name
```suggestion
          const hasInvalidPathItems =
```

**Review by clarani — COMMENTED** [review]
LGTM :white_check_mark:

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 16]
```diff
@@ -230,7 +241,38 @@ export const usePathfinding = (
...
+            } else {
+              dispatch(
+                setFailure(
+                  castErrorToFailure(pathfindingResult, {
+                    name: t('pathfinding_failed'),
+                    message: t('missing_pathSteps'),
+                  })
+                )
+              );
+            }
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Maybe you can move this in a function handleInvalidPathItems ?

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 17]
```diff
@@ -208,6 +221,51 @@ export const usePathfinding = (
     }
   }, [origin?.id, destination?.id, rollingStock]);
 
+  const handleInvalidPathItems = (
+    pathfindingResult: PathfindingResult,
+    pathSteps: (PathStep | null)[],
+    dispatch: AppDispatch,
+    pathfindingDispatch: React.Dispatch<PathfindingAction>,
+    t: TFunction
+  ) => {
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: As you are in the component, you can simplify the arguments of your function:
```suggestion
  const handleInvalidPathItems = (
    pathSteps: (PathStep | null)[],
    invalidPathItems: Extract<PathfindingInputError, { error_type: 'invalid_path_items' }>['items']
  ) => {
```

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 18]
```diff
@@ -229,8 +287,13 @@ export const usePathfinding = (
...
             pathfindingResult.failed_status === 'pathfinding_not_found' &&
             pathfindingResult.error_type === 'incompatible_constraints';
-
-          if (pathfindingResult.status === 'success' || incompatibleConstraintsCheck) {
+          if (
+            pathfindingResult.status === 'failure' &&
+            pathfindingResult.failed_status === 'pathfinding_input_error' &&
+            pathfindingResult.error_type === 'invalid_path_items'
+          ) {
+            handleInvalidPathItems(pathfindingResult, pathSteps, dispatch, pathfindingDispatch, t);
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Don't give the pathfindingResult, but directly the invalid pathSteps: `pathfindingResult.items`
```suggestion
            handleInvalidPathItems(pathSteps, pathfindingResult.items);
```

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 19]
```diff
@@ -208,6 +221,51 @@ export const usePathfinding = (
...
+    pathSteps: (PathStep | null)[],
+    dispatch: AppDispatch,
+    pathfindingDispatch: React.Dispatch<PathfindingAction>,
+    t: TFunction
+  ) => {
+    if (
+      pathfindingResult.status === 'failure' &&
+      pathfindingResult.failed_status === 'pathfinding_input_error' &&
+      pathfindingResult.error_type === 'invalid_path_items'
+    ) {
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Since you have the invalidItems in the arguments of your function, you can remove this if
```suggestion
```

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 20]
```diff
@@ -208,6 +221,51 @@ export const usePathfinding = (
...
+      //TODO: we currently only handle invalid pathSteps with trigram. We will have to do it for trackOffset, opId and uic too.
+      const invalidPathItems = pathfindingResult.items
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: You can rename this props
```suggestion
      const invalidTrigrams = pathfindingResult.items
```

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 21]
```diff
@@ -208,6 +221,51 @@ export const usePathfinding = (
...
+        pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
+        pathfindingDispatch({ type: 'PATHFINDING_PARAM_CHANGED' });
+      } else {
+        dispatch(
+          setFailure(
+            castErrorToFailure(pathfindingResult, {
+              name: t('pathfinding_failed'),
+              message: t('missing_pathSteps'),
+            })
+          )
```
Comment (clarani) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: You can directly setFailure with a name and a message :)
```suggestion
         setFailure({
            name: t('pathfinding_failed'),
            message: t('missing_pathSteps'),
          })
```

**Review by clarani — COMMENTED** [review]
You can squash and rebase :)

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 16 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 17 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 18 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 19 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 20 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/src/modules/pathfinding/hooks/usePathfinding.ts (hunk 21 above)
Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: done

File: front/public/locales/en/operationalStudies/manageTrainSchedule.json [hunk 22]
```diff
@@ -79,6 +79,7 @@
   "launchPathFinding": "Launch pathfinding",
   "launchSimulation": "Start",
   "manageVias": "Steps management",
+  "missing_pathSteps": "At least one pathstep wasn't found in the infrastructure",
```
Comment (clarani) [inline front/public/locales/en/operationalStudies/manageTrainSchedule.json]: We usually use CamelCase, even though I find snake case more readable, especially for translations :/
```suggestion
  "missingPathSteps": "At least one pathstep wasn't found in the infrastructure",
```

**Review by clarani — COMMENTED** [review]
LGTM :white_check_mark: Can you rebase and clean your commits ? :) I will approve once you have cleaned it

File: front/public/locales/en/operationalStudies/manageTrainSchedule.json (hunk 22 above)
Comment (Caracol3) [reply front/public/locales/en/operationalStudies/manageTrainSchedule.json]: done

File: front/package.json [hunk 23]
```diff
@@ -151,6 +151,9 @@
     "vite-tsconfig-paths": "^5.0.1",
     "vitest": "^2.0.0"
   },
+  "optionalDependencies": {
+    "@rollup/rollup-linux-arm64-gnu": "^4.23.0"
```
Comment (emersion) [inline front/package.json:155]: Could this be split into its own commit? It's unrelated to the bug this PR fixes.

File: front/src/modules/pathfinding/hooks/usePathfinding.ts [hunk 24]
```diff
@@ -256,9 +302,11 @@ export const usePathfinding = (
               // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
               const updatedPathSteps: (PathStep | null)[] = pathSteps.map((step, i) => {
                 if (!step) return step;
+                const trigramValue = 'trigram' in step ? step.trigram : 'no trigram found';
                 const correspondingOp = suggestedOperationalPoints.find(
                   (suggestedOp) =>
-                    'uic' in step && suggestedOp.uic === step.uic && suggestedOp.ch === step.ch
+                    ('uic' in step && suggestedOp.uic === step.uic && suggestedOp.ch === step.ch) ||
+                    (suggestedOp.trigram === trigramValue && suggestedOp.ch === step.ch)
```
Comment (emersion) [inline front/src/modules/pathfinding/hooks/usePathfinding.ts]: Instead of adding more cases in this function, could we use `matchPathStepAndOp()`? That function already handles all of the possible cases.

Comment (Caracol3) [reply front/src/modules/pathfinding/hooks/usePathfinding.ts]: Indeed ! Much clearer with the function. Thanks ;-)

File: front/package.json (hunk 23 above)
Comment (emersion) [reply front/package.json:155]: Oh, I just noticed @clarani has made the same request above.

Small note: your new commit is missing the "front:" prefix.

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark: Nice job :clap:
